galera/src/replicator_smm.cpp

```diff
@@ -1470,9 +1470,24 @@ galera::ReplicatorSMM::process_conf_change(void*                    recv_ctx,
         if (st_required && app_wants_st)
         {
             // GCache::Seqno_reset() happens here
-            request_state_transfer (recv_ctx,
-                                    group_uuid, group_seqno, app_req,
-                                    app_req_len);
+            long ret =
+                request_state_transfer (recv_ctx,
+                                        group_uuid, group_seqno, app_req,
+                                        app_req_len);
+
+            if (ret < 0 || sst_state_ == SST_CANCELED)
+            {
+                // If the request was canceled due to an error at
```

Comment:
There is no way out from this error and node can't be part of cluster.
Shouldn't we simply move it to S_CLOSING state ?